### PR TITLE
[ENH] Implement image scanners for performance enhancement

### PR DESCRIPTION
**Pull request by xuyxu**
This PR implements the function on extracting all local patches from a image dataset, which is similar to `sklearn.feature_extraction.image.extract_patches_2d` but with more options, such as the kernel size, stride, padding size.

It is used in the original paper and designed to improve the performance of deep forest on image datasets. In addition, the scanner can also be used to pre-process the dataset and improve the performance of other classifiers.

After transformations, the training dataset will be in the shape `(n_samples * n_patches_per_sample, kernel_size*kernel_size)`. Notice that since `n_samples * n_patches_per_sample` is typically very large, the data type of feature values should be cast into `np.unit8` after using the scanner.

Steps:
* [ ] Add the PyTorch backend and GPU support
* [ ] Add the Numpy backend
* [ ] Add unit tests on the image scanner (**Input**: a toy image; **Test**: All patches should be exactly what we want)
* [ ] Add the documentation
* [ ] Verify the improvements after using the image scanner

### CODE REVIEW

**Comment by zzzzwj** [comment]
I'll complete the jobs of pytorch backend and GPU support.

**Comment by xuyxu** [comment]
Great @zzzzwj, I will follow your work and add the Numpy backend then ;-)

**Comment by xuyxu** [comment]
Closed via PR #17
